src/common/indexer/elastic/elastic.indexer.helper.ts

```diff
@@ -437,11 +443,17 @@ export class ElasticIndexerHelper {
       if (filter.functions[0] === '') {
         elasticQuery = elasticQuery.withMustNotExistCondition('function');
       } else {
+        const functionConditions = [];
         for (const field of filter.functions) {
-          elasticQuery = elasticQuery.withMustCondition(QueryType.Should([
-            QueryType.Match('function', field),
+          const shouldConditions = QueryType.Should([
```

Comment:
duplicated code. could have extracted a function. also, a unit test would be helpful

Reply:
New unit tests will be added today for this
I will extract this code into a function